`TestFilePath` failed

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiproxy%2Fghpr_unit_test/detail/ghpr_unit_test/47/pipeline

### 2. What did you expect to see? (Required)

No error

### 3. What did you see instead (Required)

```
--- FAIL: TestFilePath (1.27s)

    logger.go:25: 2024-02-05T11:08:49.178+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.279+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.279+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"/tmp/TestFilePath518184147/001/cfg\""}

    logger.go:25: 2024-02-05T11:08:49.280+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.280+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"/tmp/TestFilePath518184147/001/cfg\""}

    logger.go:25: 2024-02-05T11:08:49.390+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.550+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.550+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"/tmp/TestFilePath518184147/001/cfg\""}

    logger.go:25: 2024-02-05T11:08:49.564+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.666+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.666+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"./cfg\""}

    logger.go:25: 2024-02-05T11:08:49.666+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.666+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"./cfg\""}

    logger.go:25: 2024-02-05T11:08:49.678+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.780+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.780+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"./cfg\""}

    logger.go:25: 2024-02-05T11:08:49.847+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.847+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"./cfg\""}

    logger.go:25: 2024-02-05T11:08:49.849+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.950+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.951+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"/tmp/TestFilePath518184147/001/cfg\""}

    logger.go:25: 2024-02-05T11:08:49.951+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:49.951+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"/tmp/TestFilePath518184147/001/cfg\""}

    logger.go:25: 2024-02-05T11:08:49.963+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:50.065+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:50.065+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"_tmp/cfg\""}

    logger.go:25: 2024-02-05T11:08:50.065+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:50.065+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"_tmp/cfg\""}

    logger.go:25: 2024-02-05T11:08:50.084+0800	WARN	TestFilePath	failed to rewatch config file	{"error": "no such file or directory"}

    logger.go:25: 2024-02-05T11:08:50.104+0800	WARN	TestFilePath	failed to rewatch config file	{"error": "no such file or directory"}

    logger.go:25: 2024-02-05T11:08:50.124+0800	WARN	TestFilePath	failed to rewatch config file	{"error": "no such file or directory"}

    logger.go:25: 2024-02-05T11:08:50.146+0800	WARN	TestFilePath	failed to rewatch config file	{"error": "no such file or directory"}

    logger.go:25: 2024-02-05T11:08:50.164+0800	WARN	TestFilePath	failed to rewatch config file	{"error": "no such file or directory"}

    logger.go:25: 2024-02-05T11:08:50.185+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:50.185+0800	INFO	TestFilePath	config file reloaded

    logger.go:25: 2024-02-05T11:08:50.187+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:50.289+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:50.289+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"./cfg\""}

    logger.go:25: 2024-02-05T11:08:50.347+0800	INFO	TestFilePath	current config	{"cfg": {"proxy":{"addr":"0.0.0.0:6000","pd-addrs":"127.0.0.1:2379","frontend-keepalive":{"enabled":true},"backend-healthy-keepalive":{"enabled":true,"idle":60000000000,"cnt":5,"intvl":3000000000,"timeout":15000000000},"backend-unhealthy-keepalive":{"enabled":true,"idle":10000000000,"cnt":5,"intvl":1000000000,"timeout":5000000000},"graceful-close-conn-timeout":15},"api":{"addr":"0.0.0.0:3080"},"advance":{"ignore-wrong-namespace":true},"workdir":"/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/work","security":{"server-tls":{"min-tls-version":"1.2"},"server-http-tls":{"min-tls-version":"1.2"},"cluster-tls":{"min-tls-version":"1.2"},"sql-tls":{"min-tls-version":"1.2"}},"log":{"encoder":"tidb","level":"info","log-file":{"max-size":300,"max-days":3,"max-backups":3}}}}

    logger.go:25: 2024-02-05T11:08:50.348+0800	INFO	TestFilePath	config file reloaded	{"event": "WRITE         \"./cfg\""}

    config_test.go:176: 

        	Error Trace:	/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/config_test.go:176

        	            				/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/config_test.go:259

        	            				/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/manager/config/config_test.go:288

        	Error:      	Not equal: 

        	            	expected: 1

        	            	actual  : 2

        	Test:       	TestFilePath

FAIL
```

### 4. What is your version? (Required)

<!-- Paste the output of tiproxy version -->
<!-- Paste the output of tiproxyctl version if related -->
<!-- Paste the output of SELECT tidb_version()  if related -->

master

--- a/pkg/manager/config/config_test.go
+++ b/pkg/manager/config/config_test.go
@@ -7,6 +7,7 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"runtime"
 	"strings"
 	"testing"
 	"time"
@@ -161,14 +162,25 @@ func TestFilePath(t *testing.T) {
 		count  int
 	)
 
+	const (
+		noReload int = iota
+		reloadWrite
+		reloadCreate
+	)
+
 	tmpdir := t.TempDir()
-	checkLog := func(increased bool) {
-		if increased {
+	checkLog := func(reloadType int) {
+		if reloadType != noReload {
+			expectedCount := count + 1
+			// On linux, writing once will trigger 2 WRITE events. But on macOS, it only triggers once.
+			if reloadType == reloadWrite && runtime.GOOS == "linux" {
+				expectedCount++
+			}
 			var newCount int
 			require.Eventually(t, func() bool {
 				newCount = strings.Count(text.String(), "config file reloaded")
-				return newCount > count
-			}, 3*time.Second, 10*time.Millisecond, "count=%d, newCount=%d", count, newCount)
+				return newCount == expectedCount
+			}, 3*time.Second, 10*time.Millisecond, "count=%d, expected=%d, newCount=%d", count, expectedCount, newCount)
 			count = newCount
 		} else {
 			time.Sleep(100 * time.Millisecond)
@@ -195,7 +207,7 @@ func TestFilePath(t *testing.T) {
 				newlog := filepath.Join(tmpdir, "log1")
 				require.NoError(t, os.Rename(tmplog, newlog))
 				require.NoError(t, os.Remove(newlog))
-				checkLog(false)
+				checkLog(noReload)
 			},
 		},
 		{
@@ -242,13 +254,13 @@ func TestFilePath(t *testing.T) {
 			checker: func(filename string) {
 				require.NoError(t, os.RemoveAll("_tmp"))
 				// To update `count`.
-				checkLog(false)
+				checkLog(noReload)
 
 				require.NoError(t, os.Mkdir("_tmp", 0755))
 				f, err := os.Create("_tmp/cfg")
 				require.NoError(t, err)
 				require.NoError(t, f.Close())
-				checkLog(true)
+				checkLog(reloadCreate)
 			},
 		},
 		{
@@ -256,12 +268,12 @@ func TestFilePath(t *testing.T) {
 			filename: "cfg",
 			checker: func(filename string) {
 				require.NoError(t, os.Remove(filename))
-				checkLog(false)
+				checkLog(noReload)
 
 				f, err := os.Create(filename)
 				require.NoError(t, err)
 				require.NoError(t, f.Close())
-				checkLog(true)
+				checkLog(reloadCreate)
 			},
 		},
 	}
@@ -277,11 +289,11 @@ func TestFilePath(t *testing.T) {
 
 		count = 0
 		cfgmgr, text, _ = testConfigManager(t, test.filename)
-		checkLog(false)
+		checkLog(noReload)
 
 		// Test write.
 		require.NoError(t, os.WriteFile(test.filename, []byte("proxy.pd-addrs = \"127.0.0.1:2379\""), 0644))
-		checkLog(true)
+		checkLog(reloadWrite)
 
 		// Test other.
 		if test.checker != nil {

